src/core/algorithms/dc/verifier/dc_verifier.cpp

```diff
@@ -0,0 +1,514 @@
+    RegisterOption(config::kTableOpt(&input_table_));
+}
+
+void DCVerifier::MakeExecuteOptsAvailable() {
+    MakeOptionsAvailable({config::names::kDenialConstraint});
+}
+
+void DCVerifier::LoadDataInternal() {
+    data_ = model::CreateTypedColumnData(*input_table_, true);
+    input_table_->Reset();
+    relation_ = ColumnLayoutRelationData::CreateFrom(*input_table_, true);
+}
+
+unsigned long long int DCVerifier::ExecuteInternal() {
+    auto start = std::chrono::system_clock::now();
+    DC dc;
+    try {
+        dc = SplitDC(dc_string_);
+    } catch (std::exception const& e) {
+        LOG(INFO) << e.what();
+        return 0;
+    }
+
+    bool has_header = relation_->GetSchema()->GetColumns()[0].get()->GetName() != "";
+    index_offset_ = 1 + static_cast<size_t>(has_header);
+    dc = ConvertEqualities(dc);
+    std::vector<DC> res_dc = ConvertDisequalities(dc);
+    DCType dc_type = GetType(res_dc[0]);
+    auto check = [&res_dc](DCVerifier* ver, bool (DCVerifier::*mtd)(const DC&)) {
+        return std::all_of(res_dc.begin(), res_dc.end(),
+                           [&ver, &mtd](const DC& dc) { return (ver->*mtd)(dc); });
+    };
+
+    // FIX: For better perfomance methods that use range search
+    // should better use bst in case of one-dimensional equality part
+    std::unordered_map<DCType, bool (DCVerifier::*)(const DC&)> map{
+            {DCType::kAllEquality, &DCVerifier::VerifyAllEquality},
+            {DCType::kOneInequality, &DCVerifier::VerifyOneInequality},
+            {DCType::kOneTuple, &DCVerifier::VerifyOneTuple},
+            {DCType::kTwoTuples, &DCVerifier::VerifyTwoTuples},
+            {DCType::kMixed, &DCVerifier::VerifyMixed}};
+    result_ = check(this, map.at(dc_type));
+
+    auto elapsed_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::system_clock::now() - start);
+
+    return elapsed_time.count();
+}
+
+bool DCVerifier::VerifyMixed(const DC& dc) {
+    std::vector<Predicate> predicates = dc.GetPredicates();
+    std::vector<Predicate> s_predicates, t_predicates, mixed_predicates;
+    for (auto const& pred : predicates) {
+        bool l_tuple = pred.GetLeftOperand().GetTuple();
+        bool r_tuple = pred.GetRightOperand().GetTuple();
+        switch (l_tuple + r_tuple) {
+            case 0:  // s.A op s.B
+                s_predicates.push_back(pred);
+                break;
+            case 1:  // s.A op t.B
+                mixed_predicates.push_back(pred);
+                break;
+            case 2:  // t.A op t.B
+                t_predicates.push_back(pred);
+                break;
+            default:
+                assert(false);
+                __builtin_unreachable();
+        }
+    }
+
+    DC mixed_dc(mixed_predicates.begin(), mixed_predicates.end());
+    util::KDTree<point> s_tree, t_tree;
+    bool res = true;
+    std::vector<uint> all_cols = dc.GetColumnIndices();
+    for (size_t i = 0; i < data_[0].GetNumRows(); i++) {
```

Comment:
nit:
```suggestion
    for (size_t i = 0; i < data_[0].GetNumRows(); ++i) {
```
Style guide recommends using prefix operators unless you need postfix semantics